### ISSUE TITLE
When publish markdown pages with 6.0-beta2 and active deployState parameter then pages are not published correctly

### ISSUE THREAD

**Issue by qwazer** [bsorrentino/maven-confluence-plugin#158]
plugin version:
6.0-beta2

Steps to reproduce:
  1. have site.xml with markdown pages
  2. publish without config section `deployState` - all is OK
  3. publish with config section `deployState`

It render raw internal confluence storage format instead of formatted pages:

For example.
md page with content
```
{children:all=true}
```
rendered as
```
<ac:structured-macro ac:name="children" ac:schema-version="2" ac:macro-id="0614a57f-a058-41eb-97f0-ae11cc16e73e"><ac:parameter ac:name="all">true</ac:parameter></ac:structured-macro>
```

See screen shot:
![screenshot_2018-07-10_10-01-00](https://user-images.githubusercontent.com/3875925/42494191-2e034c96-8428-11e8-8293-50127030e524.png)

**Comment by bsorrentino** [comment]
Hi @qwazer I'm unable to reproduce issue ... do you still have such problem ?

**Comment by qwazer** [comment]
@bsorrentino No I cannot reproduce, it's strange.

**Comment by qwazer** [comment]
Let's close as not reproduced

**Comment by qwazer** [comment]
I reproduced it 
![screenshot_2018-07-16_16-44-05](https://user-images.githubusercontent.com/3875925/42761906-8ac70186-8917-11e8-8c64-acf7390ca367.png)

Under investigation

**Comment by bsorrentino** [comment]
could you share the `<site file>.xml` and maven plugin configuration ?

**Comment by qwazer** [comment]
Some code snippet

````
<?xml version="1.0" encoding="UTF-8"?>
<!--suppress RequiredAttributes -->
<bsc:site
        xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance'
        xmlns:bsc='https://github.com/bsorrentino/maven-confluence-plugin'
        xsi:schemaLocation='https://github.com/bsorrentino/maven-confluence-plugin https://raw.githubusercontent.com/bsorrentino/maven-confluence-plugin/master/schemas/site-schema-3.4.0.xsd'>

    <home name="E2E SDK" uri="../markdown/e2e-sdk-parent.md" ignore-variables="true">

        <child name="01. Introduction" uri="../markdown/01-introduction/introduction.md"/>
        <child name="02. Concept" uri="../markdown/02-concept/concept.md">                                                      <!--todo-->
            <attachment name="archimate-to-be-detailed.png"    uri="../../../target/site/archimate-to-be-detailed.png"/>
            <attachment name="component-x-e2e-artifacts.png"    uri="../../../target/site/component-x-e2e-artifacts.png"/>
            <attachment name="e2e-runner.png"    uri="../../../target/site/e2e-runner.png"/>
        </child>
        <child name="03. User doc" uri="../markdown/03-user_doc/index.md">
            <child name="How to run test scenarios" uri="../markdown/03-user_doc/user_howto_run.md" ignore-variables="true"/>                <!--todo-->
            <child name="Step types" uri="../markdown/dummy.md"/>                               <!--todo-->
            <child name="How to write test scenario" uri="../markdown/dummy.md"/>               <!--todo-->
            <child name="Person concepts" uri="../markdown/03-user_doc/person_concept.md"/>
            <child name="Scenario parametrization" uri="../markdown/03-user_doc/scenario_parametrization.md" ignore-variables="true"/>
            <child name="Versioning rules" uri="../markdown/03-user_doc/versioning_rules.md" ignore-variables="true"/>
        </child>
  </home>

    <label>documentation</label>
    <label>generated</label>


</bsc:site>
````

```
  <plugin>
                        <groupId>org.bsc.maven</groupId>
                        <artifactId>confluence-reporting-maven-plugin</artifactId>
                        <version>6.0-beta2</version>
                        <executions>
                            <execution>
                                <id>site-deploy</id>
                                <phase>site</phase>
                                <goals>
                                    <goal>deploy</goal>
                                </goals>
                                <configuration>
                                    <endPoint>https://confluence.int.zone/rpc/xmlrpc</endPoint>
                                    <spaceKey>~areshetnikov</spaceKey>
                                    <parentPageTitle>Anton Reshetnikov’s Home</parentPageTitle>
                                    <serverId>confluence-server-id</serverId>
                                    <!--<deployState>-->
                                        <!--<active>true</active>-->
                                    <!--</deployState>-->
                                </configuration>
                            </execution>
                        </executions>
                    </plugin>
```

**Comment by qwazer** [comment]
It look like sometimes I have exception when publishing with rendering unknown macro. After this exception all pages becomes broken.

I'll try to reproduce it more cleary some time later.
And looks it is not related to `deployState` parameter.

**Comment by bsorrentino** [comment]
ok 

could you share stacktrace ?

**Comment by qwazer** [comment]
```
[ERROR] Failed to execute goal org.bsc.maven:confluence-reporting-maven-plugin:6.0-beta2:deploy (site-deploy) on project sdk-e2e-tests-v2: error generating report: java.lang.Exception: com.atlassian.confluence.rpc.RemoteException: com.atlassian.confluence.content.render.xhtml.migration.exceptions.UnknownMacroMigrationException: The macro 'dockerregistryhost' is unknown. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.bsc.maven:confluence-reporting-maven-plugin:6.0-beta2:deploy (site-deploy) on project sdk-e2e-tests-v2: error generating report
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
Caused by: org.apache.maven.plugin.MojoExecutionException: error generating report
	at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.execute(ConfluenceDeployMojo.java:321)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 21 more
Caused by: java.lang.Exception: java.lang.Exception: com.atlassian.confluence.rpc.RemoteException: com.atlassian.confluence.content.render.xhtml.migration.exceptions.UnknownMacroMigrationException: The macro 'dockerregistryhost' is unknown.
	at org.bsc.confluence.xmlrpc.Confluence.call(Confluence.java:933)
	at org.bsc.confluence.xmlrpc.Confluence.storePage(Confluence.java:322)
	at org.bsc.confluence.xmlrpc.XMLRPCConfluenceServiceImpl.storePage(XMLRPCConfluenceServiceImpl.java:225)
	at org.bsc.confluence.ConfluenceServiceFactory$MixedConfluenceService.storePage(ConfluenceServiceFactory.java:74)
	at org.bsc.maven.confluence.plugin.AbstractConfluenceMojo.lambda$null$5(AbstractConfluenceMojo.java:393)
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:981)
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2124)
	at org.bsc.maven.confluence.plugin.AbstractConfluenceMojo.lambda$generateChild$6(AbstractConfluenceMojo.java:388)
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:981)
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2124)
	at org.bsc.maven.confluence.plugin.AbstractConfluenceMojo.generateChild(AbstractConfluenceMojo.java:386)
	at org.bsc.maven.confluence.plugin.AbstractConfluenceSiteMojo.generateChildren(AbstractConfluenceSiteMojo.java:248)
	at org.bsc.maven.confluence.plugin.AbstractConfluenceSiteMojo.generateChildren(AbstractConfluenceSiteMojo.java:256)
	at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.generateProjectReport(ConfluenceDeployMojo.java:623)
	at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.lambda$generateProjectReport$8(ConfluenceDeployMojo.java:637)
	at org.bsc.confluence.ConfluenceServiceFactory$MixedConfluenceService.call(ConfluenceServiceFactory.java:132)
	at org.bsc.maven.confluence.plugin.AbstractBaseConfluenceMojo.confluenceExecute(AbstractBaseConfluenceMojo.java:216)
	at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.generateProjectReport(ConfluenceDeployMojo.java:635)
	at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.execute(ConfluenceDeployMojo.java:306)
	... 23 more
```

**Comment by qwazer** [comment]
There is no such error, when I disable deployState config option or revert to previous verion of plugin `5.1.1`

**Comment by bsorrentino** [comment]
Thanks ...

Looks inexplicable ... I have to investigate the reasons

**Comment by qwazer** [comment]
@bsorrentino 
I prepared steps to reproduce:

Take a look at [my commit](https://github.com/qwazer/maven-confluence-plugin/commit/c265f2cb02a1cc77ba2179ef730fb969b16a18d2) 
It based on your commit 8452558cf229bbe10970864eeebac4e897e2d8db

On XML-RPC based calls command
```
mvn org.bsc.maven:confluence-reporting-maven-plugin:6.0-SNAPSHOT:deploy@issue158
```
gives error:
```
[ERROR] Failed to execute goal org.bsc.maven:confluence-reporting-maven-plugin:6.0-SNAPSHOT:deploy (issue158) on project maven-confluence-test: error generating report: java.lang.Exception: com.atlassian.confluence.rpc.RemoteException: com.atlassian.confluence.content.render.xhtml.migration.exceptions.UnknownMacroMigrationException: The macro 'my_prop' is unknown. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.bsc.maven:confluence-reporting-maven-plugin:6.0-SNAPSHOT:deploy (issue158) on project maven-confluence-test: error generating report
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
Caused by: org.apache.maven.plugin.MojoExecutionException: error generating report
	at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.execute(ConfluenceDeployMojo.java:321)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 21 more
Caused by: java.lang.Exception: java.lang.Exception: com.atlassian.confluence.rpc.RemoteException: com.atlassian.confluence.content.render.xhtml.migration.exceptions.UnknownMacroMigrationException: The macro 'my_prop' is unknown.
	at org.bsc.confluence.xmlrpc.Confluence.call(Confluence.java:933)
	at org.bsc.confluence.xmlrpc.Confluence.storePage(Confluence.java:322)
	at org.bsc.confluence.xmlrpc.XMLRPCConfluenceServiceImpl.storePage(XMLRPCConfluenceServiceImpl.java:225)
	at org.bsc.confluence.ConfluenceServiceFactory$MixedConfluenceService.storePage(ConfluenceServiceFactory.java:74)
	at org.bsc.maven.confluence.plugin.AbstractConfluenceMojo.lambda$null$5(AbstractConfluenceMojo.java:392)
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:981)
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2124)
	at org.bsc.maven.confluence.plugin.AbstractConfluenceMojo.lambda$generateChild$6(AbstractConfluenceMojo.java:387)
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:981)
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2124)
	at org.bsc.maven.confluence.plugin.AbstractConfluenceMojo.generateChild(AbstractConfluenceMojo.java:385)
	at org.bsc.maven.confluence.plugin.AbstractConfluenceSiteMojo.generateChildren(AbstractConfluenceSiteMojo.java:248)
	at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.generateProjectReport(ConfluenceDeployMojo.java:623)
	at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.lambda$generateProjectReport$8(ConfluenceDeployMojo.java:637)
	at org.bsc.confluence.ConfluenceServiceFactory$MixedConfluenceService.call(ConfluenceServiceFactory.java:132)
	at org.bsc.maven.confluence.plugin.AbstractBaseConfluenceMojo.confluenceExecute(AbstractBaseConfluenceMojo.java:216)
	at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.generateProjectReport(ConfluenceDeployMojo.java:635)
	at org.bsc.maven.confluence.plugin.ConfluenceDeployMojo.execute(ConfluenceDeployMojo.java:306)
	... 23 more
```

**Comment by bsorrentino** [comment]
Hi @qwazer  thanks for further info

It seems related  exclusively  to `xmlrpc` when skip publication 

I'll fix it soon

**Comment by bsorrentino** [comment]
I've got the test working , take a look commit [94e860f](https://github.com/bsorrentino/maven-confluence-plugin/commit/94e860f5e0ce5a76a6713ca835775d17ac090f2d)

I've deployed also  a new `6.0-SNAPSHOT'

**Comment by qwazer** [comment]
@bsorrentino yes it works now. Thanks!

**Comment by bsorrentino** [comment]
fix released in version `6.0-rc1`
